File: android/src/main/java/com/capacitorjs/plugins/sim/SimPlugin.java
package com.capacitorjs.plugins.sim;

import com.getcapacitor.JSObject;
import com.getcapacitor.Plugin;
import com.getcapacitor.PluginCall;
import com.getcapacitor.PluginMethod;
import com.getcapacitor.annotation.CapacitorPlugin;

@CapacitorPlugin(name = "Sim")
public class SimPlugin extends Plugin {

    private Sim implementation;

    @Override
    public void load() {
        implementation = new Sim(getContext());
    }

    @PluginMethod
    public void getInfo(PluginCall call) {
        JSObject res = new JSObject();

        res.put("carrierId", implementation.getSimCarrierId());
        res.put("carrierIdName", implementation.getSimCarrierIdName());
        res.put("countryIso", implementation.getSimCountryIso());
        res.put("operator", implementation.getSimOperator());
        res.put("operatorName", implementation.getSimOperatorName());

        call.resolve(res);
    }
}

File: ios/Plugin/SimPlugin.swift
import Foundation
import Capacitor

/**
 * Please read the Capacitor iOS Plugin Development Guide
 * here: https://capacitorjs.com/docs/plugins/ios
 */
@objc(SimPlugin)
public class SimPlugin: CAPPlugin {

    private let implementation: Sim = {
        return Sim()
    }()


    @objc func getInfo(_ call: CAPPluginCall) {
        call.resolve([
          "carrierId": implementation.getSimCarrierId(),
          "carrierIdName": implementation.getSimCarrierIdName(),
          "countryIso": implementation.getSimCountryIso(),
          "operator": implementation.getSimOperator(),
          "operatorName": implementation.getSimOperatorName(),
        ])
    }
}
